api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CommonCohortQueries.java

```diff
@@ -262,7 +262,7 @@ public CohortDefinition getMohMQPatientsOnCondition(
       query += "AND p.gender = 'F' ";
     }
     query +=
-        "AND e.encounter_datetime >= :startDate AND e.encounter_datetime <= :endDate "
+        "AND o.obs_datetime >= :startDate AND o.obs_datetime <= :endDate "
```

Comment:
why you changed to obs_datetime?

Reply:
According to Eude, the patient 22022 has "Diagnostico TB" = "Continua" and was being excluded from B3 while should be included in the B3 query,  and then be excluded on the CompositionString, reason why he recommended the use of obs_datetime instead of  encounter_datetime.

Reply:
but adulto_segmente uses  encountre_datetime?

Reply:
The Spec say so, please, take a look.

Reply:
where the spec says we have to use obs_datetime?